compatibility/bazel_tools/daml_ledger/Sandbox.hs

```diff
@@ -74,16 +75,20 @@ getSandboxProc SandboxConfig{..} portFile = do
                 , "--crt", sandboxCertificates </> "server.crt"
                 ]
         else pure []
-    pure $ proc sandboxBinary $ concat
-        [ sandboxArgs
-        , [ "--port=0", "--port-file", portFile ]
-        , tlsArgs
-        , [ timeArg ]
-        , [ "--client-auth=" <> clientAuthArg auth | Just auth <- [mbClientAuth] ]
-        , [ "--auth-jwt-hs256-unsafe=" <> secret | Just secret <- [mbSharedSecret] ]
-        , [ "--ledgerid=" <> ledgerId | Just ledgerId <- [mbLedgerId] ]
-        , dars
-        ]
+    let args = concat
+          [ sandboxArgs
+          , [ "--port=0", "--port-file", portFile ]
+          , tlsArgs
+          , [ timeArg ]
+          , [ "--client-auth=" <> clientAuthArg auth | Just auth <- [mbClientAuth] ]
+          , [ "--auth-jwt-hs256-unsafe=" <> secret | Just secret <- [mbSharedSecret] ]
+          , [ "--ledgerid=" <> ledgerId | Just ledgerId <- [mbLedgerId] ]
+          , dars
+          ]
+    env <- getEnvironment
+    pure $ (proc sandboxBinary args)
+      -- Reducing memory consumption to allow multiple parallel test executions.
+      { env = Just $ ("_JAVA_OPTIONS", "-Xss4m -Xms128m -Xmx1g") : env }
```

Comment:
I tried `-Xss2m` at first (which is the default in `@daml//bazel_tools:scala.bzl`). But, I observed issues with the sandbox startup on my machine that went away with 4M.